net/tcp/tcp_input.c

```diff
@@ -1178,7 +1178,7 @@ static void tcp_input(FAR struct net_driver_s *dev, uint8_t domain,
                   return;
                 }
             }
-          else
+          else if ((conn->tcpstateflags & TCP_STATE_MASK) <= TCP_ESTABLISHED)
```

Comment:
```suggestion
          else
            {
#ifdef CONFIG_NET_TCP_OUT_OF_ORDER
              /* Queue out-of-order segments. */
              tcp_input_ofosegs(dev, conn, iplen);
#endif
              if ((conn->tcpstateflags & TCP_STATE_MASK) <= TCP_ESTABLISHED)
                {
                  tcp_send(dev, conn, TCP_ACK, tcpiplen);
                }
              return;
```

Reply:
need goto Lin1220 and processed case TCP_FIN_WAIT ,so can't return in this case. what about this?:
 ```
          else
            {
#ifdef CONFIG_NET_TCP_OUT_OF_ORDER
              /* Queue out-of-order segments. */
              tcp_input_ofosegs(dev, conn, iplen);
#endif
              if ((conn->tcpstateflags & TCP_STATE_MASK) <= TCP_ESTABLISHED)
                {
                  tcp_send(dev, conn, TCP_ACK, tcpiplen);
                  return;
                }
           }
```

Reply:
Yes, you are right, we should include the return

Reply:
Done